app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt

```diff
@@ -158,6 +180,186 @@ class StateFragmentTest {
+          R.id.state_recycler_view,
+          0,
+          R.id.interaction_button
+        )
+      ).check(matches(withText(R.string.state_continue_button)))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExplorationTest5_previousAndNextButtonIsNotDisplayed() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.previous_state_image_view)).check(
+        matches(
+          not(
+            isDisplayed()
+          )
+        )
+      )
+      onView(
+        atPositionOnView(
+          R.id.state_recycler_view,
+          0,
+          R.id.next_state_image_view
+        )
+      ).check(matches(not(isDisplayed())))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExplorationTest5_submitAnswer_clickContinueButton_previousButtonIsDisplayed() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.interaction_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.interaction_button)).perform(click())
+      onView(
+        atPositionOnView(
+          R.id.state_recycler_view,
+          0,
+          R.id.previous_state_image_view
+        )
+      ).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExplorationTest5_submitAnswer_clickContinueButton_clickPreviousButton_previousButtonIsHiddenAndNextButtonIsDisplayed() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.interaction_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.interaction_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.previous_state_image_view)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.previous_state_image_view)).check(
+        matches(
+          not(
+            isDisplayed()
+          )
+        )
+      )
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.next_state_image_view)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExplorationTest5_submitAnswer_clickContinueButton_submitAnswer_clickPreviousButton_clickNextButton_continueButtonIsDisplayed() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      onView(withId(R.id.play_exploration_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.interaction_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.interaction_button)).perform(click())
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.interaction_button)).perform(click())
```

Comment:
Why does this need to be clicked three times here but only two times earlier?

Reply:
Actually this corresponds to the special test-case mentioned in #257 . Also, the ordering of clicks and everything is correct. I have added comments to make it more clear.